### ISSUE TITLE
WebSocket server w/o SSL not working?

### ISSUE THREAD

**Issue by mfranke23** [civetweb/civetweb#1078]
Dear civetweb devs,

**tldr;**
On "master", the new `ws_server` example doesn't compile, "undefined symbols" for `EVP_Digest` etc. And in our application that embeds civetweb, I can only get websockets to work when SSL is enabled.

Is this work-in-progress or what is the plan for using websockets without SSL?

**Details**

Compiling the new `ws_server` example to work with the supplied `build.sh` fails with "undefined symbols" for EVP_Digest etc:

```
../../src/civetweb.c:13005:2: warning: implicit declaration of function 'EVP_Digest' is invalid in C99 [-Wimplicit-function-declaration]
        EVP_Digest((unsigned char *)buf, (uint32_t)strlen(buf), (unsigned char *)sha,
        ^
../../src/civetweb.c:13006:9: warning: implicit declaration of function 'EVP_get_digestbyname' is invalid in C99 [-Wimplicit-function-declaration]
                NULL, EVP_get_digestbyname("sha1"), NULL);
                      ^
3 warnings generated.Undefined symbols for architecture x86_64:
  "_EVP_Digest", referenced from:
      _send_websocket_handshake in civetweb-943e87.o
  "_EVP_get_digestbyname", referenced from:
      _send_websocket_handshake in civetweb-943e87.o
ld: symbol(s) not found for architecture x86_64

```

Similarly, in our application a websocket handshake crashes when SSL is not init'ed because EVP_Digest is a null pointer.

The problem seems to be the [`EVP_Digest`](https://github.com/civetweb/civetweb/blob/4447b6501d5c568b4c6c0940eac801ec690b2250/src/civetweb.c#L13005) call in `send_websocket_handshake()` which is only available when the ssl context is set up. (Up to v1.15, the required SHA1 function was taken from `sha1.inl` but that was replaced with `EVP_Digest`.)

**Comment by bel2125** [comment]
This is an unintended side effect of this recent change:
https://github.com/civetweb/civetweb/commit/6bf1a495ea73fde53355bac2e06ac32e0afd3dec
https://github.com/civetweb/civetweb/pull/1072

**Comment by mfranke23** [comment]
Alright, thanks for the quick reply!

We'll go with the v1.15 release code for the time being, so issue can be closed.
